scripts/classification/imagenet/train_imagenet.py

```diff
@@ -322,21 +333,16 @@ def train(ctx):
                 lam = np.random.beta(opt.mixup_alpha, opt.mixup_alpha)
                 if epoch >= opt.num_epochs - opt.mixup_off_epoch:
                     lam = 1
-                data_mixup = [lam*X + (1-lam)*X[::-1] for X in data]
+                data = [lam*X + (1-lam)*X[::-1] for X in data]
 
-                label_mixup = []
                 if opt.label_smoothing:
                     eta = 0.1
                 else:
                     eta = 0.0
-                for Y in label:
-                    y1 = label_transform(Y, classes, eta)
-                    y2 = label_transform(Y[::-1], classes, eta)
-                    label_mixup.append(lam*y1 + (1-lam)*y2)
+                label = mixup_transform(label, classes, lam, eta)
 
-                data = data_mixup
-                label = label_mixup
             elif opt.label_smoothing:
+                hard_label = label
                 label = smooth(label, classes)
```

Comment:
So this `label` is not used now? After adding line 362 the `train_metric` seems to be updated with `hard_label`, irrelevant to the smoothed `label`.

Reply:
`label` is used in loss calculation.